src/Workspaces/Core/Portable/SymbolId/SymbolKey.cs

```diff
@@ -6,173 +6,151 @@
-    ///   <item>Two IPropertySymbol's are the "same" if they have the "same" containing type, the same
-    ///     <see cref="ISymbol.MetadataName"/>,  and have same parameter types and <see cref="IParameterSymbol.RefKind"/>.</item>
-    ///   <item>Two ITypeParameterSymbol's are the "same" if they have the "same" containing symbol and the <see cref="ISymbol.MetadataName"/>.</item>
-    ///   <item>Two IFieldSymbol's are the "same" if they have the "same" containing symbol and the <see cref="ISymbol.MetadataName"/>.</item>
+    ///   <item>Two <see cref="IParameterSymbol"/>s are the "same" if they have
+    ///         the "same" <see cref="ISymbol.ContainingSymbol"/> and 
+    ///         equal <see cref="ISymbol.MetadataName"/>.</item>
+    ///   <item>Two <see cref="IPointerTypeSymbol"/>s are the "same" if they have 
+    ///         the "same" <see cref="IPointerTypeSymbol.PointedAtType"/>.</item>
+    ///   <item>Two <see cref="IPropertySymbol"/>s are the "same" if they have 
+    ///         the "same" the "same" <see cref="ISymbol.ContainingType"/>, 
+    ///         the "same" <see cref="ISymbol.MetadataName"/>, and have 
+    ///         the "same" <see cref="IParameterSymbol.Type"/>s and  
+    ///         the "same" <see cref="IParameterSymbol.RefKind"/>s.</item>
+    ///   <item>Two <see cref="ITypeParameterSymbol"/> are the "same" if they have
+    ///         the "same" <see cref="ISymbol.ContainingSymbol"/> and 
+    ///         the "same" <see cref="ISymbol.MetadataName"/>.</item>
+    ///   <item>Two <see cref="IFieldSymbol"/>s are the "same" if they have
+    ///         the "same" <see cref="ISymbol.ContainingSymbol"/> and 
+    ///         the "same" <see cref="ISymbol.MetadataName"/>.</item>
     /// </list>    
-    /// A SymbolID for an IAliasSymbol will <see cref="SymbolKey.Resolve"/> back to the ISymbol for
-    /// the <see cref="IAliasSymbol.Target"/>.
-    /// </para>
     /// <para>
-    /// Due to issues arising from errors and ambiguity, it's possible for a SymbolKey to resolve to
-    /// multiple symbols. For example, in the following type:
-    ///  <code>
-    /// class C
-    /// {
-    ///    int Foo();
-    ///    bool Foo();
-    /// }
-    /// </code>
-    /// The SymbolKey for both Foo methods will be the same.  The SymbolId will then resolve to both methods.
+    ///     Due to issues arising from errors and ambiguity, it's possible for a SymbolKey to resolve to
+    ///     multiple symbols. For example, in the following type:
+    ///     <code>
+    ///     class C
+    ///     {
+    ///        int M();
+    ///        bool M();
+    ///     }
+    ///     </code>
+    ///     The SymbolKey for both 'M' methods will be the same.  The SymbolKey will then resolve to both methods.
+    /// </para>
     /// </para>
     /// </summary>
-    internal abstract partial class SymbolKey
+    internal partial class SymbolKey
```

Comment:
I considered this.  However, it would  mean having to compute and store the hashcode up front (otherwise, it wouldn't be copied around when people copied the struct).

However, there are multiple hash codes to compute.  The one when you do care about assembly ID, and the one where you don't.  

Alternatively, i can just compute and store the hashcode in the form where we don't care about the assembly id.  It would mean more collusions, but that's probably not an issue in practice.